src/fonduer/parser/parser.py

```diff
@@ -141,6 +141,11 @@ def apply(  # type: ignore
             progress_bar=progress_bar,
         )
 
+    def _add(self, doc: Union[Document, None]) -> None:
```

Comment:
Do we use this function?

Reply:
`Parser._add` is called in the main process after getting a result from `ParserUDF.apply` at
https://github.com/HazyResearch/fonduer/blob/5909011e9bdc52825e480f941a0dff4737002b8e/src/fonduer/utils/udf.py#L143-L150

The `UDF._add` has been meant to add features/labels to the database, more specifically by 

https://github.com/HazyResearch/fonduer/blob/5909011e9bdc52825e480f941a0dff4737002b8e/src/fonduer/features/featurizer.py#L248-L252

and 

https://github.com/HazyResearch/fonduer/blob/5909011e9bdc52825e480f941a0dff4737002b8e/src/fonduer/supervision/labeler.py#L309-L311

.
So only `Featurizer._add` and `Labeler._add` have been implemented, but I realized that `Parser._add` can be implemented to store the "transient" `Document` to database.
We don't need `_add` for `MentionExtractor` and `CandidateExtractor` as extracted mentions and candidates will be persisted without explicitly adding them to the session.